brambling/forms/organizer.py

```diff
@@ -167,6 +152,116 @@ def save(self):
+    template_event = forms.ModelChoiceField(queryset=Event.objects.all(), required=False)
+
+    class Meta:
+        model = Event
+        fields = ('name', 'slug', 'start_date', 'end_date', 'start_time',
+                  'end_time', 'organization', 'template_event')
+
+    def __init__(self, request, *args, **kwargs):
+        super(EventCreateForm, self).__init__(*args, **kwargs)
+        self.request = request
+        if not request.user.is_authenticated():
+            raise ValueError("EventCreateForm requires an authenticated user.")
+        self.fields['template_event'].queryset = Event.objects.filter(
+            Q(organization__owner=request.user) |
+            Q(organization__editors=request.user)
+        ).order_by('-last_modified').distinct()
+        self.fields['organization'].queryset = Organization.objects.filter(
+            Q(owner=request.user) |
+            Q(editors=request.user)
+        ).order_by('name').distinct()
+
+    def clean(self):
+        cd = super(EventCreateForm, self).clean()
+        if ('start_date' in cd and 'end_date' in cd and
+                cd['start_date'] > cd['end_date']):
+            self.add_error('start_date', "Start date must be before or equal to the end date.")
+
+        if (cd.get('template_event') and cd.get('organization') and
+                cd['template_event'].organization_id != cd['organization'].id):
+            self.add_error('template_event', "Template event and new event must be from the same organization.")
+
+        if 'slug' in cd and cd.get('organization'):
+            if Event.objects.filter(organization=cd['organization'], slug=cd['slug']).exists():
+                self.add_error('slug', 'Slug is already in use by another event; please choose a different one.')
+
+        return cd
+
+    def _adjust_date(self, old_event, new_event, date):
+        """
+        Returns a date relative to the new event's start date as the given date
+        is relative to the olde event's start date.
+
+        """
+        return date + (new_event.start_date - old_event.start_date)
+
+    def save(self):
+        if self.instance.is_demo():
+            self.instance.api_type = Event.TEST
+
+        self.instance.application_fee_percent = self.instance.organization.default_application_fee_percent
+        instance = super(EventCreateForm, self).save()
+
+        template = self.cleaned_data.get('template_event')
+        if template:
+            fields = (
+                'description', 'website_url', 'banner_image', 'city',
+                'state_or_province', 'country', 'timezone', 'currency',
+                'has_dances', 'has_classes', 'liability_waiver', 'privacy',
+                'collect_housing_data', 'collect_survey_data', 'cart_timeout',
+                'check_postmark_cutoff', 'transfers_allowed', 'facebook_url',
+            )
+            for field in fields:
+                setattr(instance, field, getattr(template, field))
+
```

Comment:
Damn, yeah, the related objects are copied but not the core data, because it's not saved yet. :-p